### PR TITLE
Lido Finance

### PR DESCRIPTION
**Pull request by fortysevenlabs**
Validation
* [Subgraph link for validation](https://thegraph.com/hosted-service/subgraph/fortysevenlabs/lido)

To review:
- [x] Rewards are minted in stETH. Do we want to calculate revenues with stETH price ([see this](https://github.com/messari/subgraphs/compare/master...fortysevenlabs:messari-subgraphs:master#diff-f425ab52dd0bdc8463cd9d36191e6cfe4e116c6e5ad90fd09a63da2e8b4f462dR165))
- [x] YearnlensOracle token prices accuracy (~currently commented out~, minor differences, ok to use, re-enabled)

### CODE REVIEW

File: subgraphs/lido/src/entityUpdates/financialMetrics.ts [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+  amount: BigInt
+): void {
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  const protocol = getOrCreateProtocol();
+
+  // inputToken is ETH, price with ETH
+  const amountUSD = bigIntToBigDecimal(amount).times(
+    getOrCreateToken(Address.fromString(ETH_ADDRESS), block.number)
+      .lastPriceUSD!
+  );
```
Comment (harsh9200) [inline subgraphs/lido/src/entityUpdates/financialMetrics.ts:31]: unused variable

File: subgraphs/lido/src/entityUpdates/financialMetrics.ts [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+  // Pool
+  pool.cumulativeSupplySideRevenueUSD = pool.cumulativeTotalRevenueUSD.minus(
+    pool.cumulativeProtocolSideRevenueUSD
+  );
+  pool.save();
+
+  // Pool Daily
+  poolMetricsDailySnapshot.cumulativeSupplySideRevenueUSD =
+    pool.cumulativeSupplySideRevenueUSD;
+  poolMetricsDailySnapshot.dailySupplySideRevenueUSD = poolMetricsDailySnapshot.dailyTotalRevenueUSD.minus(
```
Comment (harsh9200) [inline subgraphs/lido/src/entityUpdates/financialMetrics.ts:232]: Some of the values in `dailySupplySideRevenueUsd` is zero ([Link](https://subgraphs.messari.io/subgraph?endpoint=https%3A%2F%2Fapi.thegraph.com%2Fsubgraphs%2Fname%2Ffortysevenlabs%2Flido&tab=pool&poolId=0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84&view=poolDailySnapshots-dailySupplySideRevenueUSD)). Can you add a condition here to only update the field when the revenue is positive?

**Review by harsh9200 — COMMENTED** [review]
Great Work! As for the revenue, it should be calculated in `stETH` instead of `ETH`, it will be more accurate and logically correct.

**Comment by fortysevenlabs** [comment]
> As for the revenue, it should be calculated in stETH instead of ETH, it will be more accurate and logically correct.
Exactly, thanks for the confirmation, will continue to price using ETH for input/deposit but price the output/stakingRewards in stETH for accuracy.

**Comment by fortysevenlabs** [comment]
> As for the revenue, it should be calculated in stETH instead of ETH, it will be more accurate and logically correct.
Exactly, thanks for the confirmation, will continue to price using ETH for input/deposit but price the output/stakingRewards in stETH for accuracy.

File: subgraphs/lido/src/entityUpdates/financialMetrics.ts (hunk 2 above)
Comment (fortysevenlabs) [reply subgraphs/lido/src/entityUpdates/financialMetrics.ts]: Pushed the change.

**Comment by harsh9200** [comment]
> Actually, thinking further about this. Since the rewards are issued to the protocol from ETH2 deposit contract in the form of ETH (which is then used to rebase the user with stETH shares), we should be fine calculating revenues using ETH. stETH would have been necessary if we were tracking user's balances/value.

Sounds Good, LGTM!

**Comment by this-username-is-taken** [comment]
Great work! Would you be able to add CI/CD to it? See here for details: https://github.com/messari/subgraphs/tree/master/deployment

**Comment by this-username-is-taken** [comment]
Also, if you could update Lido here: https://github.com/messari/subgraphs#development-status as well

**Comment by fortysevenlabs** [comment]
Will add/update:
- [ ] CI/CD
- [ ] Dev status

**Comment by this-username-is-taken** [comment]
Thanks Tarun!

**Comment by fortysevenlabs** [comment]
What's the merge and deploy plan?

Are you or your team validating [the graph](https://subgraphs.messari.io/subgraph?endpoint=https://api.thegraph.com/subgraphs/name/fortysevenlabs/lido-ethereum) as well?

**Comment by fortysevenlabs** [comment]
Cool, that was fast, thanks for the merge 👍

**Pull request merged**